### ISSUE TITLE
Inverse of multiplication

### ISSUE THREAD

**Issue by llllvvuu** [wacabanga/reverseflow#13]
I think I figured out how to make FFT for deconvolution (known filter, since unknown filter is a bit more annoying) work in a straightforward manner: it'll be deterministic given a working (pointwise) multiplication inverse. Do we have a good multiplication inverse though? 0 is annoying (as an output, or as a known input).

**Comment by zenna** [comment]
Scalar multiplication?
So the parametric inverse we have in the code f(z;theta) = (theta, z/theta) has two problems:
1. undefined at theta=0 (not a problem in practice)
2. incomplete when z=0, e.g. we can get back the pair (5,0) but never (0,5)

In the paper we also have
f(z; theta) = (z^(theta), z^(1-theta))
1. Is broken at zero
2. can't produce negative values and hence is incomplete.
3. Will probably be numerically unstable

**Comment by zenna** [comment]
Do you see any other problems?

One somewhat hacky approach would be to extend the first parametric inverse which a kind of switch

f(z; Θ1 ∈ R, Θ2 ∈ {0, 1}) = if Θ2 then (Θ1, z/Θ1) else (z/Θ1, Θ1)

This is not ideal because (i) we have more parameters than degrees of freedom, (ii) the parameter is boolean and hence bad for optimization, (iii) the parametric inverse is no longer bijective

**Comment by zenna** [comment]
My hunch is we just need a special case for zero.

The question is what are the implications for stability when we are dealing with floating point values and not real numbers.

**Comment by llllvvuu** [comment]
Yeah scalar multiplication. These are basically my thoughts as well
